### PR TITLE
feat: Bundle CDT-LSP plugins with Espressif-IDE

### PR DESCRIPTION
**Pull request by kolipakakondal**
This PR enable the following.
1. Bundle CDT LSP plugin with the IEP update site 
2. Bundle CDT LSP plugin in the Espressif-IDE
3. Auto configure clangd path
4. With new project create .clangd file with compiler args

## Description

Please include a summary of the change and which issue is fixed.

https://jira.espressif.com:8443/browse/IEP-1117
https://jira.espressif.com:8443/browse/IEP-1149

## Type of change

- New feature (non-breaking change which adds functionality)

## How has this been tested?

- Downloaded PR update site and verified CDT LSP editor preference options
- Downloaded PR Espressif-IDE and verified LSP editor.

Screenshot 2023-12-26 at 5 34 11 PM

Screenshot 2023-12-26 at 5 39 01 PM

https://github.com/espressif/idf-eclipse-plugin/blob/master/docs/clangd_cdt_support.md

**Test Configuration**:
* ESP-IDF Version: 5.1
* OS (Windows,Linux and macOS):macOS

## Dependent components impacted by this PR:

- Update site
- C/C+ editor

## Checklist
- [ ] PR Self Reviewed
- [ ] Applied Code formatting
- [ ] Added Documentation
- [ ] Added Unit Test
- [ ] Verified on all platforms - Windows,Linux and macOS

## Summary by CodeRabbit

- **New Features**
    - Enhanced toolchain management with the ability to find compilers based on target names.
    - Improved project configuration and build process in Eclipse IDE for IoT Development Framework (IDF) projects.
    - Added support for updating `.clangd` configuration files in IDF projects, enabling better integration with Language Server Protocol (LSP).
    - Expanded the IoT Development Framework (IDF) environment with new dependencies and bundles to enhance development capabilities.

- **Documentation**
    - Updated project settings and configurations for Eclipse IDE users, including Java compiler settings and version control ignore settings.

- **Chores**
    - Updated plugin dependencies and configurations to support new features and improve the development experience in the Eclipse IDE environment.

### CODE REVIEW

File: bundles/com.espressif.idf.lsp/.classpath
Comment (alirana01) [inline bundles/com.espressif.idf.lsp/.classpath:1]: I think we need to change a lot of our classpath files to have Java 17 you can see the warnings when building. 
We can have a separate ticket for this

**Review by alirana01 — COMMENTED** [review]
Looks very good. A solid effort indeed. Can we please try to add the code formatter for the new project as well.

**Comment by AndriiFilippov** [comment]
@kolipakakondal hi!

Tested under:
OS - Windows 10
ESP-IDF: v5.1.2

1. the Driver Path does not recognize gcc location. 
image



2. no colored text highlighting
image

**Comment by AndriiFilippov** [comment]
@kolipakakondal hi ! 

Windows 10 


open workspace -> go to "Preferences" - "Editor LSP" (clangd) -> empty Clangd Path 
image

Now install tools ->  go to "Preferences" - "Editor LSP" (clangd) -> Path hasn't been updated. Eclipse Restart fix issue. 

image

**Comment by kolipakakondal** [comment]
> @kolipakakondal hi!
> 
> Tested under: OS - Windows 10 ESP-IDF: v5.1.2
> 
> 1. the Driver Path does not recognize gcc location.
>    image
> 2. no colored text highlighting
>    image

fixed

**Comment by AndriiFilippov** [comment]
> > @kolipakakondal hi!
> > Tested under: OS - Windows 10 ESP-IDF: v5.1.2
> > 
> > 1. the Driver Path does not recognize gcc location.
> >    image
> > 2. no colored text highlighting
> >    image
> 
> fixed

@kolipakakondal the issues is still persist. 

For example, if user does not have tools in .espressif folder installed - then is still shows empty PATH even after installation. 
image


Please, delete .espressif folder and follow steps: 

open workspace -> go to "Preferences" - "Editor LSP" (clangd) -> **empty Clangd Path** -> Now install tools -> go to "Preferences" - "Editor LSP" (clangd) -> **Still Empty PATH**
Eclipse Restart fix issue.

**Pull request merged**